elixir/lib/app_identity/validation.ex

```diff
@@ -108,6 +114,18 @@ defmodule AppIdentity.Validation do
     {:error, "padlock must be a string value"}
   end
 
+  def validate(:padlock, "") do
+    {:error, "padlock must not be an empty string"}
+  end
+
+  def validate(:padlock, value) do
+    if Regex.match?(~r/^[a-fA-F0-9]+$/, value) do
```

Comment:
I much prefer `~r/^[[:xdigit:]]+$/ (even if the POSIC spelling of “hexdigit” irritates me)